src/Client/Ollama.php

```diff
@@ -256,6 +255,72 @@ private function parse(string $data): \Generator {
     }
   }
 
+  /**
+   * Parse data received from Ollama into string array.
+   *
+   * The data from Ollama when request in stream may not be complete JSON
+   * objects superheated by new-lines. It may be partial JSON objects, and the
+   * objects theme self may contain new-lines (as the result from the LLM itself
+   * uses new-lines). So simply splitting on new-lines is not a solution.
+   *
+   * Also, the first and last parts of the data may be partial objects or string
+   * that will be completed in the next response.
```

Comment:
So the data from Ollama is ndjson https://en.wikipedia.org/wiki/JSON_streaming#Newline-delimited_JSON, but the objects themselves may also contain newlines?

Reply:
yes, see the example data in this PR description... it's a mix of the different cases seen returned. In f.eks. python you can request the next line from a stream, but in php with PSR7 interface you request x bytes (e.g. 1024). Which results in the data being broken into random chuncks... that hash to be glue together correctly to get json objects and at the same time deliverer data as fast as possible to the frontend to give the "live chat" effect...